[docs] Remove out-of-date docs, improve header standardization. (#24456)
microsoft/vcpkg: docs/examples/overlay-triplets-linux-dynamic.md

# Overlay triplets example

> `vcpkg` comes with many useful officially supported configurations in the box, including dynamic and static libraries for Windows as well as static libraries for MacOS and Linux. There are also a large number of community-supported triplets available, such as Emscripten or Android.

Reviewer: I'm not sure I like this added paragraph; if we're selling triplets the concept IMO that should be in the doc talking about what triplets are rather than repeated in a sample?

Author: I have reverted changes to this file to better scope this PR.

> For more advanced customization, users can provide their own triplets via the Overlay Triplets mechanism. See our [triplets documentation](../users/triplets.md) for an exhaustive reference of what can be configured in a triplet file.

Reviewer: ```suggestion
For more advanced customization, you can provide your own triplets via the Overlay Triplets mechanism. See our [triplets documentation](../users/triplets.md) for an exhaustive reference of what can be configured in a triplet file.
```

I think we should only use "users" when the ultimate customer isn't the immediate reader of the documentation, if that makes sense.

Author: See my below comment.

Reviewer: I think your below comment applies to all my comments *except* this one. This paragraph is entirely new.
